### PR TITLE
sam4l: Update Watchdog

### PR DESCRIPTION
**Pull request by ppannuto**
### Pull Request Overview

This updates the WDT to use StaticRef and the use the Osc32 (like the AST) to allow low-power operation.

### Testing Strategy

TODO

### Documentation Updated

- [x] ~~Kernel: Updated the relevant files in `/docs`, or no updates are required.~~
- [x] ~~Userland: Added/updated the application README, if needed.~~

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by ppannuto** [comment]
Well, that proved more work than expected. This now works correctly though, including when the core goes into deep sleep. Should be good to go.

Tested on hail + hail app. Simple testing:
```diff
$ git diff
diff --git a/boards/hail/src/main.rs b/boards/hail/src/main.rs
index aba76ac6..e5dbf5e1 100644
--- a/boards/hail/src/main.rs
+++ b/boards/hail/src/main.rs
@@ -496,5 +496,6 @@ pub unsafe fn reset_handler() {
         &mut PROCESSES,
         FAULT_RESPONSE,
     );
+    (&sam4l::wdt::WDT as &kernel::hil::watchdog::Watchdog).start(2000);
     kernel::main(&hail, &mut chip, &mut PROCESSES, &hail.ipc);
 }
```

**Comment by alevy** [comment]
Oh this is going to be very useful for debugging that bug Brad noticed where process timers seem to stop working after a while with more than one app. Because I _believe_ it has to do with taking a long time to wake up from deep sleep (~2ms), it's been tricky to debug.

File: chips/sam4l/src/wdt.rs [hunk 1]
```diff
@@ -93,86 +95,127 @@ register_bitfields![u32,
...
+
+        // When writing to the affected fields, the user must ensure a wait
+        // corresponding to 2 clock cycles of both the WDT peripheral bus clock
+        // and the selected WDT clock source.
+        //
+        // TODO: Actual math based on chosen clock, ASF does:
+        //       delay = div_ceil(sysclk_hz(), OSC_[chosen]_NOMINAL_HZ)
+        unsafe {
+            for _ in 0..10000 {
+                asm!("nop");
```
Comment (alevy) [inline chips/sam4l/src/wdt.rs]: we have a `kernel::support::nop()` function which is marked safe, so you can use that and get rid of the `unsafe`.

**Comment by ppannuto** [comment]
Good call, fixed

**Pull request merged**